source/_components/climate.markdown

```diff
@@ -20,6 +20,39 @@ climate:
+  * The **target_temp_high** property selects an upper limit on the acceptable temperature. The device will start cooling (if `cooling` or `auto` are selected as `action`) if that temperature is exceeded.
+  * The **target_temp_low** property selects a lower limit on the acceptable temperature. The device will start heating (if `heating` or `auto` are selected as `action`) if that temperature is exceeded.
+* The **target_humidity** property selects the target humidity. If this is set and the device is able to dry / humidify the air, it will do so if the actual humidity differs from this value.
+* The **control** property specifies how the device is currently being controlled. Many devices allow to chose between an automated control (following a given preset, see below) or a manual control. *Note*: This does *not* influence whether the device is controlled by Home Assistant. The device is always controlled by Home Assistant, even if in `manual` mode. This property controls only the setting on the actual device. Choices are:
+ * Preset (device follows the device-specific preset, see below)
+ * Manual (the device itself is only controlled manually / via Home Assistant)
+* The **source** property specifies (for devices which support this), what source to use for achieving the goal given in the `action`. Choices are:
+ * Electric
+ * Gas
+ * Fan_only
+* The **preset** property controls which preset to follow, if `preset` is selected for the `control` property. If the devices is in `manual` mode, this setting has no effect. The choices for this property are not standardized but differ from device to device. Note that the presets must be provided by the device, either statically or by direct configuration of the device. Home Assistant does not yet allow you to create / edit presets. If you want to emulate a preset, an automation and a timer component might be what you want.
+* The **away** property can only be on or off and sets devices into a away-mode in which they usually conserve energy by heating / cooling less.
+* The **fan_mode** property selects between the different modes of the device's fan, if it has any. Choices differ from device to device.
+* The **swing_mode** property selects whether and how the device should swing, if it supports this. Choices differ from device to device.
+
+Additionally, there are a couple of deprecated properties that are still implemented by older platforms, but will gradually be replaced by the properties mentioned above:
+
+* The **operation_mode** property (**deprecated**) is a mix of `action` and `control` above.
```

Comment:
After looking through various implementation (some of which I find wrong), I think operation_mode makes sense. 
operation_mode should be determined by the component and specify what types of function could your "climate" component perform.

Some climate devices can do only heating, some can do only cooling, some can do both, some can do both and switch automatically between them, and some should have "off" which would ignore any other commands.

Reply:
That's why I suggest the `action` property. It's basically just a change in name (and some semantics) for two reasons:
* To make clear what the property is used for (I think 'action' is a more clear wording than 'operation_mode'). While "economy" might be an `operation_mode`, it should be pretty clear that "economy" is not an `action` (but rather a `preset`).
* To have a clear separation between the old set of attributes and the new set of attributes.

Reply:
I have to agree, that a change would allow for easier refactoring. Also I really like the "preset" idea. That would move people away from hacking operation_modes and fiddling in weird ways with "hold" modes.